File: old/core/aggregators/services/header_element/context.py
"""
Header element extraction context.

Dataclass for passing context to header element extraction strategies.
"""

from dataclasses import dataclass

from ..image_store import build_image_ref


@dataclass
class HeaderElementContext:
    """Context for header element extraction strategies."""

    url: str  # Source URL
    alt: str  # Alt text for image/title for iframe
    user_id: int | None = None  # Optional user ID for authenticated API calls


@dataclass
class HeaderElementData:
    """Data returned from header element extraction strategies."""

    image_bytes: bytes  # Raw image data
    content_type: str  # MIME type (e.g. 'image/jpeg')
    content_hash: str  # SHA-256 of the stored (compressed) bytes
    image_url: str | None = None  # Original image URL for removal from content

    @property
    def image_ref(self) -> str:
        """The ``yana-img://`` reference callers render as the image src."""
        return build_image_ref(self.content_hash)

File: src/lib/aggregators/header/context.ts
import { buildImageRef } from "../images/store";

/**
 * Context passed to header element extraction strategies.
 */
export interface HeaderElementContext {
  url: string;
  alt?: string;
  userId?: number | null;
  onLog?: (message: string) => void;
  /**
   * The article page's HTML, when the caller has already fetched it.
   *
   * `FullWebsiteAggregator.enrichArticles()` fetches every article page for
   * its content anyway, and the og:image/page-image strategies want that same
   * page -- so without this the aggregator fetched each article twice, once
   * here and once for the content, doubling the request count against every
   * site it aggregates. Optional because not every caller has it: the
   * `article.reload` job and the RSS-only aggregators reach header extraction
   * with nothing fetched yet, and those still fall back to fetching the page.
   */
  html?: string;
}

/**
 * Data returned from header element extraction strategies.
 */
export interface HeaderElementData {
  imageBytes: Buffer;
  contentType: string;
  contentHash: string;
  imageUrl?: string | null;
}

/**
 * Return the `yana-img://` reference for a HeaderElementData object.
 */
export function getHeaderImageRef(data: HeaderElementData): string {
  return buildImageRef(data.contentHash);
}
